jerry-core/ecma/builtin-objects/ecma-builtin-regexp-prototype.c

```diff
@@ -89,39 +89,45 @@ ecma_builtin_regexp_prototype_compile (ecma_value_t this_arg, /**< this argument
 
         /* Get source. */
         ecma_string_t *magic_string_p = ecma_get_magic_string (LIT_MAGIC_STRING_SOURCE);
-        ecma_property_t *prop_p = ecma_op_object_get_property (target_p, magic_string_p);
-        pattern_string_p = ecma_get_string_from_value (ecma_get_named_data_property_value (prop_p));
+        ecma_value_t source_value = ecma_op_object_get_own_data_prop (target_p, magic_string_p);
         ecma_deref_ecma_string (magic_string_p);
+        pattern_string_p = ecma_get_string_from_value (source_value);
+        ecma_deref_ecma_string (pattern_string_p);
```

Comment:
That is a tricky code. The pattern_string_p is defined in the outside scope and wants to be a reference to an existing string depending on various conditions. Since target_p still has the reference to the original string, dereferencing this string does not destroy it. This logic was in the original code.